src/btree/bt_split.c

```diff
@@ -1183,6 +1174,7 @@ err:	if (!complete)
 			if (next_ref->state == WT_REF_SPLIT)
 				next_ref->state = WT_REF_DELETED;
 		}
+	WT_TRET(__wt_fair_unlock(session, &parent->pg_intl_split_lock));
```

Comment:
Self review - this shouldn't be here. I've removed it from #2213 - I'll wait for a decision as to whether that branch is a preferred solution before fixing it here.